macleann/enchiridion-server
# Fix #13 local episodes being stored with incorrect id's

# Fix local episodes being stored with incorrect id's

- Removed unnecessary episode fixtures
- Changed pk's on all episode fixtures
- Added `tmdb_id` and `series_id` keys to all episode fixtures
- Changed `episode_id`'s on all playlistepisode fixtures
- Added `tmdb_id` and `series_id` fields to `models/episode.py`
- Added those same two fields to `LocalEpisodeSerializer` in `serializers/episode_serializer.py`
- Removed migrations, deleted local `db.sqlite3` file, created a new `db.sqlite3` file, and reran the `seed_db.sh` script to make new migrations, migrate, and load all updated fixture data.
- Updated `create` and `update` functions in `user_playlist_view.py` to include the `tmdb_id` and `series_id` fields in the retrieved episode's data if the episode wasn't already stored locally and thus needed to be retrieved from TMDB.


Fixes #13 local episodes being stored with incorrect id's

## Type of change


# How to Test?

Make sure you've followed all installation and setup instructions in the README

```
git fetch origin nm-fixing-episode-ids
git checkout nm-fixing-episode-ids
python manage.py runserver
```

- [ ] Open Postman and send a `GET` request to `http://localhost:8000/playlists/1`
- [ ] Send a `PUT` request to `http://localhost:8000/playlists/1` with the following:
```
{
    "name": "Essential Season 2 Episodes Only",
    "description": "Only the episodes that ultimately tie into the show's overall plot",
    "episodes": [
        {
            "season_number": 2,
            "episode_number": 1,
            "order_number": 1,
            "series_id": 15260
        },
        {
            "season_number": 2,
            "episode_number": 8,
            "order_number": 2,
            "series_id": 15260
        },
        {
            "season_number": 2,
            "episode_number": 10,
            "order_number": 3,
            "series_id": 15260
        },
        {
            "season_number": 2,
            "episode_number": 11,
            "order_number": 4,
            "series_id": 15260
        },
        {
            "season_number": 2,
            "episode_number": 12,
            "order_number": 5,
            "series_id": 15260
        },
        {
            "season_number": 2,
            "episode_number": 13,
            "order_number": 6,
            "series_id": 15260
        },
        {
            "season_number": 2,
            "episode_number": 14,
            "order_number": 7,
            "series_id": 15260
        },
        {
            "season_number": 2,
            "episode_number": 15,
            "order_number": 8,
            "series_id": 15260
        },
        {
            "season_number": 2,
            "episode_number": 16,
            "order_number": 9,
            "series_id": 15260
        },
        {
            "season_number": 2,
            "episode_number": 17,
            "order_number": 10,
            "series_id": 15260
        },
        {
            "season_number": 2,
            "episode_number": 18,
            "order_number": 11,
            "series_id": 15260
        },
        {
            "season_number": 2,
            "episode_number": 19,
            "order_number": 12,
            "series_id": 15260
        },
        {
            "season_number": 2,
            "episode_number": 20,
            "order_number": 13,
            "series_id": 15260
        },
        {
            "season_number": 2,
            "episode_number": 21,
            "order_number": 14,
            "series_id": 15260
        },
        {
            "season_number": 2,
            "episode_number": 22,
            "order_number": 15,
            "series_id": 15260
        }
    ]
}
```
- [ ] You should get a valid response from the server

# Checklist:
